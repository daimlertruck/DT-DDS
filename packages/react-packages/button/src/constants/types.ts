import { ComponentSize } from '@dt-ui/react-core';

export type ButtonVariant = 'solid' | 'outlined' | 'text';

export type ButtonColor = 'primary' | 'error' | 'secondary';

export type ButtonAppearance = ButtonColor | 'contrast';

export type ButtonSize = Extract<ComponentSize, 'medium' | 'large' | 'small'>;
