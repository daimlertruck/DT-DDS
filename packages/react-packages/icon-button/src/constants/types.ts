export type IconButtonVariant = 'contrast' | 'default';
