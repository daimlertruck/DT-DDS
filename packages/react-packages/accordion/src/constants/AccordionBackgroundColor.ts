export const AccordionBackgroundColor = {
  White: 'white',
  Grey: 'grey',
} as const;

export type AccordionBackgroundColor =
  (typeof AccordionBackgroundColor)[keyof typeof AccordionBackgroundColor];
