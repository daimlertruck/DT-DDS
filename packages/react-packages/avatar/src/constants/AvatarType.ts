export const AvatarType = {
  Primary: 'primary',
  Secondary: 'secondary',
  Tertiary: 'tertiary',
  Profile: 'profile',
} as const;

export type AvatarType = (typeof AvatarType)[keyof typeof AvatarType];
