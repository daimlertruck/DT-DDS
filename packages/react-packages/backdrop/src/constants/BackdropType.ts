export const BackdropType = {
  Dark: 'dark',
  Light: 'light',
} as const;

export type BackdropType = (typeof BackdropType)[keyof typeof BackdropType];
