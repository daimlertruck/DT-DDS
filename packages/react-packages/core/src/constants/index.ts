export * from './z-index';
export * from './colors';
