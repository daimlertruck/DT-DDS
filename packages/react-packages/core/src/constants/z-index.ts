export const TABLE_STICKY_COLUMN_Z_INDEX = 1;
export const TABLE_HEADER_Z_INDEX = 2;
export const DROPDOWN_MENU_Z_INDEX = 3;
export const TOOLTIP_Z_INDEX = 11;
export const BACKDROP_Z_INDEX = 9;
export const MODAL_Z_INDEX = 10;
export const DRAWER_Z_INDEX = 9;
export const TOAST_Z_INDEX = 10;
