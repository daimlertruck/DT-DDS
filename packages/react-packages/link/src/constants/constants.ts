import { IconProps } from '@dt-dds/react-icon';

import { LinkSize } from './types';

export const iconSize: Record<LinkSize, IconProps['size']> = {
  small: 'xs',
  medium: 's',
  large: 'l',
};
