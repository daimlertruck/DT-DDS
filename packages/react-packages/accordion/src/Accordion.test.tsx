import { withProviders } from '@dt-ui/react-core';
import { render, screen, fireEvent } from '@testing-library/react';

import { AccordionBackgroundColor } from './constants';

import { Accordion } from '.';

describe('<ProvidedAccordion /> component', () => {
  const ProvidedAccordion = withProviders(Accordion);

  it('renders correctly the closed accordion', () => {
    const { container } = render(
      <ProvidedAccordion headerContent='Header Content' isOpen={false}>
        <div>Accordion Body</div>
      </ProvidedAccordion>
    );

    expect(container).toMatchSnapshot();
  });

  it('renders correctly the opened accordion', () => {
    render(
      <ProvidedAccordion headerContent='Header Content' isOpen>
        <div>Body Content</div>
      </ProvidedAccordion>
    );

    expect(screen.getByText('Header Content')).toBeInTheDocument();
    expect(screen.getByText('Body Content')).toBeInTheDocument();
  });

  it('toggles open/close when clicking the header', () => {
    render(
      <ProvidedAccordion headerContent='Header Content' isOpen={false}>
        Accordion Body
      </ProvidedAccordion>
    );

    const header = screen.getByTestId('accordion-header');

    fireEvent.click(header);
    expect(screen.getByTestId('accordion-body')).toHaveStyle('display: block');

    fireEvent.click(header);
    expect(screen.getByTestId('accordion-body')).toHaveStyle('display: none');
  });

  it('applies correct background color', () => {
    render(
      <ProvidedAccordion
        backgroundColor={AccordionBackgroundColor.Grey}
        headerContent='Header Content'
        isOpen={false}
      >
        Accordion Body
      </ProvidedAccordion>
    );

    const accordionContainer = screen.getByTestId('accordion-container');
    expect(accordionContainer).toHaveStyle(
      'background-color: rgb(222, 222, 222);'
    );
  });

  it('applies showIcon prop as false', () => {
    render(
      <ProvidedAccordion headerContent='Header Content' isOpen={false}>
        Accordion Body
      </ProvidedAccordion>
    );

    expect(screen.queryByTestId('accordion-icon')).toBeNull();
  });

  it('applies isDisabled prop', () => {
    render(
      <ProvidedAccordion
        headerContent='Header Content'
        isDisabled
        isOpen={false}
      >
        Accordion Body
      </ProvidedAccordion>
    );

    const accordionContainer = screen.getByTestId('accordion-container');
    expect(accordionContainer).toHaveStyle(
      'background-color: rgb(243, 243, 243);'
    );
  });
});
