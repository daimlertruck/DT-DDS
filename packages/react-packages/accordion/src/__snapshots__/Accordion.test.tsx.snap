// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProvidedAccordion /> component renders correctly the closed accordion 1`] = `
.emotion-0 {
  position: relative;
  width: 100%;
  min-width: '248px';
  background-color: #FFFFFF;
  border-radius: 0px;
  box-shadow: 0px 1px 0px 0px #EBECED;
}

.emotion-0:hover {
  background-color: #ECF7FB;
  box-shadow: 0px 1px 2px 0px rgba(0, 0, 0, 0.05);
}

.emotion-2 {
  padding: 16px;
  cursor: pointer;
}

@media (min-width: 600px) {
  .emotion-2 {
    padding: 24px;
  }
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-6 {
  width: 100%;
}

.emotion-8 {
  width: 24px;
  height: 24px;
  margin-left: 12px;
  color: #A1A2A3;
}

.emotion-10 {
  padding: 16px;
  padding-top: 0;
  display: none;
}

@media only screen and (min-width: 600px) {
  .emotion-10 {
    padding: 24px;
    padding-top: 0;
  }
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="accordion-container"
  >
    <header
      aria-disabled="false"
      class="emotion-2 emotion-3"
      data-testid="accordion-header"
      role="button"
      tabindex="0"
    >
      <div
        class="emotion-4 emotion-5"
      >
        <div
          class="emotion-6 emotion-7"
        >
          Header Content
        </div>
        <div
          class="emotion-8 emotion-9"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M11.9999 5.83L15.1699 9L16.5799 7.59L11.9999 3L7.40991 7.59L8.82991 9L11.9999 5.83ZM11.9999 18.17L8.82991 15L7.41991 16.41L11.9999 21L16.5899 16.41L15.1699 15L11.9999 18.17Z"
                fill="currentColor"
                id="Vector"
              />
            </g>
          </svg>
        </div>
      </div>
    </header>
    <div
      class="emotion-10 emotion-11"
      data-testid="accordion-body"
    >
      <div>
        Accordion Body
      </div>
    </div>
  </div>
</div>
`;
